internal/transport/http2_client.go

```diff
@@ -1266,7 +1266,7 @@ func (t *http2Client) operateHeaders(frame *http2.MetaHeadersFrame) {
 		// that the peer is speaking gRPC and we are in gRPC mode.
 		isGRPC         = !initialHeader
 		mdata          = make(map[string][]string)
-		contentTypeErr string
+		contentTypeErr = "missing content-type"
```

Comment:
Actually, why not do something similar with http status?  I.e.

```go
httpStatusErr := ""
if !endStream {
	httpStatusErr = "malformed header: missing HTTP status"
}
...
	case ":status":
		if hf.Value == "200" {
			httpStatusErr = ""
			break
		}
		httpStatusErr = fmt.Sprintf(....)
...
// Delete constructHTTPErrMsg since it will just be a simple strings.Join()
```

Reply:
Done